redis/src/caching/config.rs

```diff
@@ -0,0 +1,60 @@
+use std::num::NonZeroUsize;
+use std::time::Duration;
+
+/// Defines CacheMode of CacheConfig
+#[derive(Clone, PartialEq, Debug, Copy)]
+pub enum CacheMode {
+    /// Try to cache every command.
+    All,
+    /// Only try to cache commands which explicitly called cache methods.
+    /// If OptIn is used `cache` or `cache_with_client_side_ttl` functions
+    /// must be used on [`crate::Cmd`] for caching to work with  the command.
+    OptIn,
+}
+
+/// Options for managing client side caching
+#[derive(Clone, Debug, Copy)]
+pub struct CacheConfig {
+    pub(crate) mode: CacheMode,
+    pub(crate) size: NonZeroUsize,
+    pub(crate) default_client_ttl: Duration,
+}
+
+impl CacheConfig {
+    /// Creates new CacheConfig with default values (10_000 LRU size and 30 minutes of client ttl)
+    pub fn new() -> Self {
+        Self {
+            mode: CacheMode::All,
+            size: NonZeroUsize::new(10_000).unwrap(),
+            default_client_ttl: Duration::from_secs(60 * 30),
+        }
+    }
+
+    /// Sets how client side caching should work, default is [`crate::caching::CacheMode::All`].
+    pub fn set_mode(mut self, mode: CacheMode) -> Self {
+        self.mode = mode;
+        self
+    }
+
+    /// Sets maximum key count for the cache.
+    pub fn set_size(mut self, size: NonZeroUsize) -> Self {
+        self.size = size;
+        self
+    }
+
+    ///  Sets the default client side TTL when it's not set with `cache_with_client_side_ttl` per command.
```

Comment:
try to avoid acronyms in documentation, or at least add the expanded version - 
Sets the default client-side Time To Live (TTL) for cached values, when a TTL isn't explicitly passed on a `Cmd`.